### PR TITLE
Usertwopointohh

### PR DESCRIPTION
**Pull request by Obult**
Changes:

- reworked gateService
    - split in two providers for the same service
- add statusService (online, offline, in-game)
- add chat leaveRoom endpoint

### CODE REVIEW

**Comment by Obult** [comment]
http://localhost:3000/status/test
enpoint om in de backend de gebruikers per socket te printen

File: backend/volume/src/game/game.service.ts [hunk 1]
```diff
@@ -69,8 +69,8 @@ export class CurrentGameState {
 export class GameService {
   constructor(private readonly prismaGameService: PrismaGameService,
               private readonly prismaUserService: PrismaUserService,
-              private readonly gate: GameGateService,) {
-  }
+              @Inject('gameGate') private readonly gameGate: GateService,
```
Comment (rvan-mee) [inline backend/volume/src/game/game.service.ts:72]: why name no longer gate, is issue ty

**Review by rvan-mee — APPROVED** [review]
Ready to ruuble

**Review by rvan-mee — APPROVED** [review]
Ready to ruuble

File: backend/volume/src/game/game.service.ts (hunk 1 above)
Comment (Obult) [reply backend/volume/src/game/game.service.ts:72]: these get imported by statusService which needs to differentiate between chat- and gameGate, name is kept same for consistency

**Pull request merged**